### ISSUE TITLE
prvkey to pubkey

### ISSUE THREAD

**Issue by hamnaz** [matja/bitcoin-tool#44]
when converting 20mil hex privatekey to pubkeys, take 3 hours, when pubkeys to address, take only 16 minutes
why prvkey to pubkey take toooo mych time, any solution, or you have plan to update these tools to use cuda, where aspected works would be in min/sec 
hope see your updates regarding this issue

**Comment by streamofstars** [comment]
Can you give example of the code you use?

**Comment by hamnaz** [comment]
./bitcoin-tool --input-type private-key --input-format hex --output-type public-key --output-format hex --input-file xaa1.txt --batch  --network bitcoin --public-key-compression compressed

**Comment by streamofstars** [comment]
I tested it too and need to confirm the results @hamnaz reports, took my computer 2,5h to convert 20m hex prvkeys to pubkeys, and 3 minutes to convert to addresses in the next step.

Until any improvements are added, I can suggest splitting the input file into multiple smaller ones and running multiple instances of the bitcoin-tool on them concurrently. It is far from CUDA performance but as an ad hoc solution will probably do - workaround like this can be easily scripted, and if you have multiple cores CPU you will cut the time needed for processing considerably.

**Comment by Meru852** [comment]
Can i use to find the private key by the address?

**Comment by Meru852** [comment]
Can i use to find the private key by the address?

**Comment by streamofstars** [comment]
No, it's not possible to reverse hashing. Theoretically you could use this tool to just generate all possible addresses and compare them to the address you interested in but it is not feasible as it would take you billions of years, even using cloud, CUDA and whatever best supercomputer there is.
